sdk/tables/azure-data-tables/inc/azure/data/tables/models.hpp

```diff
@@ -22,6 +24,93 @@ namespace Azure { namespace Data { namespace Tables {
   class TableServiceClient;
   class TableClient;
 
+  /**
+   * @brief API version for Tables service.
+   */
+  class ServiceVersion final {
+  public:
+    /**
+     * @brief Construct a new Service Version object
+     *
+     * @param version The string version for Tables Service.
+     */
+    explicit ServiceVersion(std::string version) : m_version{std::move(version)} {}
+
+    /**
+     * @brief Enable comparing between two versions.
+     *
+     * @param other Another service version to be compared.
+     */
+    bool operator==(const ServiceVersion& other) const { return m_version == other.m_version; }
+
+    /**
+     * @brief Enable comparing between two versions.
+     *
+     * @param other Another service version to be compared.
+     */
+    bool operator!=(const ServiceVersion& other) const { return !(*this == other); }
+
+    /**
+     * @brief Returns string representation.
+     *
+     */
+    std::string const& ToString() const { return m_version; }
+
+    /**
+     * @brief API version 2019-12-12.
+     *
+     */
+    AZ_DATA_TABLES_DLLEXPORT const static ServiceVersion V2019_02_02;
+
+  private:
+    std::string m_version;
+  };
+
+  /**
+   * @brief Audiences available for Blobs
+   *
+   */
+  class TablesAudience final
+      : public Azure::Core::_internal::ExtendableEnumeration<TablesAudience> {
+  public:
+    /**
+     * @brief Construct a new TablesAudience object
+     *
+     * @param tablesAudience The Azure Active Directory audience to use when forming authorization
+     * scopes. For the Language service, this value corresponds to a URL that identifies the Azure
+     * cloud where the resource is located. For more information: See
+     * https://learn.microsoft.com/en-us/azure/storage/blobs/authorize-access-azure-active-directory
+     */
+    explicit TablesAudience(std::string tablesAudience)
+        : ExtendableEnumeration(std::move(tablesAudience))
+    {
+    }
+  };
+
+  /**
+   * @brief Optional parameters for constructing a new TableClient.
+   */
+  struct TableClientOptions final : Azure::Core::_internal::ClientOptions
```

Comment:
I would recommend either:
1) keeping this in the table_client.hpp file OR
2) move this to a separate table_client_options.hpp file.

It doesn't fit within `models.hpp`.

See attestation_client_options.hpp, certificate_client_options.hpp, blob_options.hpp, keyvault_options, etc. for precedence.